lib/spark.js

```diff
@@ -253,17 +254,23 @@ export default class Spark {
         this.handleRunError(err)
       }
       const duration = Date.now() - started
-      const delay = calculateDelayBeforeNextTask({
-        roundLengthInMs: APPROX_ROUND_LENGTH_IN_MS,
-        maxJitterInMs: MAX_JITTER_BETWEEN_TASKS_IN_MS,
-        maxTasksPerRound: this.#tasker.maxTasksPerRound,
-        lastTaskDurationInMs: duration,
-      })
+      const isHealthy = this.#activity.isHealthy()
+
+      let delay
+      if (isHealthy) {
+        delay = calculateDelayBeforeNextTask({
+          roundLengthInMs: APPROX_ROUND_LENGTH_IN_MS,
+          maxJitterInMs: MAX_JITTER_BETWEEN_TASKS_IN_MS,
+          maxTasksPerRound: this.#tasker.maxTasksPerRound,
+          lastTaskDurationInMs: duration,
+        })
+      } else {
+        delay = OFFLINE_RETRY_DELAY_MS
+      }
```

Comment:
Please move this logic into `calculateDelayBeforeNextTask`. It will make the function `run()` simpler to reason about, and it will be easier to test this new functionality without mocks.

```suggestion
      let delay
      const delay = calculateDelayBeforeNextTask({
        roundLengthInMs: APPROX_ROUND_LENGTH_IN_MS,
        maxJitterInMs: MAX_JITTER_BETWEEN_TASKS_IN_MS,
        maxTasksPerRound: this.#tasker.maxTasksPerRound,
        lastTaskDurationInMs: duration,
        isHealthy,
      })
```